src/mpi/coll/allreduce/allreduce_intra_recursive_multiplying.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include <math.h>
+
+#include "mpiimpl.h"
```

Comment:
I am not sure about the compilation errors. Could you try swap the order of the above two include files? `mpiimpl.h` includes the configure headers and configure headers need be included before most posix headers because it contains macros that may affect the parsing.

Reply:
I switched them!

Reply:
Could you check does this happen to fix the issue in #7176?

Reply:
Good call, that fixed it. I closed the issue. I dug into the jenkins logs and found the warnings it is complaining about. I will address those

Reply:
The compiler warnings have been eliminated